### PR TITLE
Improve implementation.

### PR DESCRIPTION
**Pull request by chrono-meter**
Please see changes.

### CODE REVIEW

File: vix/VixHost.py [hunk 1]
```diff
@@ -94,29 +98,58 @@ def __init__(self, service_provider=VIX_SERVICEPROVIDER_DEFAULT, host=None, cred
...
+            if isinstance(host, collections.Sequence) and len(host) == 2 and host[1] != 0:
+                host = ('%s:%s' % host, 0)
+
+            elif host is None and service_provider == self.VIX_SERVICEPROVIDER_VMWARE_WORKSTATION_SHARED:
+                if platform.system() == 'Windows':
+                    vmhostd_proxy_xml = os.path.join(os.getenv('ProgramData'), 'VMware', 'hostd', 'proxy.xml')
+                elif platform.system() == 'Linux':
+                    vmhostd_proxy_xml = '/etc/vmware/hostd/proxy.xml'
+                else:
+                    raise NotImplemented('Unrecognized OS or architecure ({0}, {1}, {2})'.format(platform.architecure(), platform.machine(), platform.system()))
```
Comment (naim94a) [inline vix/VixHost.py:111]: What about Darwin?

File: vix/VixHost.py [hunk 2]
```diff
@@ -1,6 +1,10 @@
 from __future__ import absolute_import
 
-from collections import namedtuple
+import collections
+import platform
+import os
+import os.path
+from xml.etree import ElementTree
```
Comment (naim94a) [inline vix/VixHost.py:7]: lxml is not in setup.py, I would suggest to add it or use a built in module.

File: vix/VixBackend.py [hunk 3]
```diff
@@ -141,38 +141,46 @@ def __init__(self):
...
 
-			return base_path
+            try:
+                import winreg
+            except ImportError:
+                import _winreg as winreg
 
-		raise NotImplementedError('Unrecognized OS or architecture ({0}, {1}, {2})'.format(platform.architecture(), platform.machine(), platform.system()))
+            if platform.machine().lower() == 'amd64':
+                base_path = winreg.QueryValueEx(winreg.OpenKey(winreg.HKEY_LOCAL_MACHINE, r'SOFTWARE\WOW6432Node\VMware, Inc.\VMware VIX'), 'InstallPath')[0]
```
Comment (naim94a) [inline vix/VixBackend.py]: Very nice 😃,
I think it's worth adding a try: except FileNotFound, PermissionError and fallback to default paths.

File: vix/VixBackend.py [hunk 4]
```diff
@@ -141,38 +141,46 @@ def __init__(self):
...
-			base_path = os.path.join(base_path, 'VMware', 'VMware VIX')
-			base_path = os.path.join(base_path, 'Vix64AllProductsDyn.dll' if arch == '64bit' else 'VixAllProductsDyn.dll')
+        elif os_name == 'Windows':
+            # Py: 32; Machine: 32: %ProgramFiles%\VMware\VMware VIX\VixAllProductsDyn.dll
+            # Py: 32; Machine: 64: %ProgramFiles(x86)%\VMware\VMware VIX\VixAllProductsDyn.dll
+            # Py: 64; Machine: 32: N/A
+            # Py: 64; Machine: 64: %ProgramFiles(x86)%\VMware\VMware VIX\Vix64AllProductsDyn.dll
 
-			return base_path
+            try:
```
Comment (naim94a) [inline vix/VixBackend.py]: winreg is python3, _winreg is python2: Please move to _compat.py

**Review by naim94a — CHANGES_REQUESTED** [review]

File: vix/VixHost.py (hunk 2 above)
Comment (chrono-meter) [reply vix/VixHost.py:7]: xml.etree.ElementTree is in [The Python Standard Library after Python 2.5](https://docs.python.org/2/library/xml.etree.elementtree.html).

File: vix/VixHost.py (hunk 1 above)
Comment (chrono-meter) [reply vix/VixHost.py:111]: I don't have macOS system. But there is no feature like *shared virtual machines* in VMware Fusion.

In the [VMware Fusion 8 Documentation](https://pubs.vmware.com/fusion-8/index.jsp?topic=%2Fcom.vmware.fusion.using.doc%2FGUID-30FCA4B3-D9FD-40AF-8817-F0902AE6D758.html):

> Shared virtual machines in Fusion work differently than shared virtual machines in Workstation Pro. In Workstation Pro, a shared virtual machine is a virtual machine on the host system that remote Workstation Pro users can access as a remote virtual machine. In Fusion, shared virtual machines can be accessed only on the local Mac host.

File: vix/VixBackend.py (hunk 4 above)
Comment (chrono-meter) [reply vix/VixBackend.py]: What about [six](https://pypi.python.org/pypi/six)?

Comment (naim94a) [reply vix/VixBackend.py]: Sure you can use six - just add to setup.py & requirements.txt...

**Comment by chrono-meter** [comment]
I think fixed all problems.

**Review by naim94a — APPROVED** [review]

**Pull request merged**